src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    let newState = { ...currentState };
```

Comment:
There's no need to create a new copy of the state at the beginning of each iteration. You can directly modify `currentState` since it's already a copy of the original state and is not being used elsewhere after modification.